src/libs/actions/PaymentMethods.js

```diff
@@ -136,14 +141,34 @@ function makeDefaultPaymentMethod(password, bankAccountID, fundID, previousPayme
                     errors: null,
                 },
             },
+            {
+                onyxMethod: CONST.ONYX.METHOD.MERGE,
+                key: previousPaymentMethod.accountType === CONST.PAYMENT_METHODS.BANK_ACCOUNT ? ONYXKEYS.BANK_ACCOUNT_LIST : ONYXKEYS.CARD_LIST,
+                value: {
+                    [previousPaymentMethod.methodID]: {
+                        ...previousPaymentMethod,
+                        isDefault: false,
+                    },
+                },
+            },
+            {
+                onyxMethod: CONST.ONYX.METHOD.MERGE,
+                key: currentPaymentMethod.accountType === CONST.PAYMENT_METHODS.BANK_ACCOUNT ? ONYXKEYS.BANK_ACCOUNT_LIST : ONYXKEYS.CARD_LIST,
+                value: {
+                    [currentPaymentMethod.methodID]: {
+                        ...currentPaymentMethod,
+                        isDefault: true,
+                    },
+                },
+            },
         ],
         failureData: [
             {
                 onyxMethod: CONST.ONYX.METHOD.MERGE,
                 key: ONYXKEYS.USER_WALLET,
                 value: {
-                    walletLinkedAccountID: previousPaymentMethodID,
-                    walletLinkedAccountType: previousPaymentMethodType,
+                    walletLinkedAccountID: previousPaymentMethod.methodID,
+                    walletLinkedAccountType: previousPaymentMethod.accountType,
                 },
             },
         ],
```

Comment:
You're also going to have to reset the `isDefault` prop in for the current and previous payment method, correct? Right now, if setting a default payment method fails (eg. the user enters the wrong password), then the previousPaymentMethod still has isDefault = false and the currentPaymentMethod has isDefault = true.

Reply:
Userwallet doesn't have the `isDefault` property.

<img width="462" alt="Screenshot 2022-09-27 at 3 16 49 PM" src="https://user-images.githubusercontent.com/26106822/192566559-40a88cb3-465d-44a5-87ad-38991d7a91f2.png">

Reply:
Great catch!